bokehjs/src/lib/models/tools/gestures/range_tool.ts

```diff
@@ -122,10 +122,35 @@ export class RangeTool extends Tool {
 
   update_ranges_from_overlay(): void {
     const {left, right, top, bottom} = this.overlay
-    if (this.x_range != null && this.x_interaction)
-      this.x_range.setv({start: left, end: right})
-    if (this.y_range != null && this.y_interaction)
-      this.y_range.setv({start: bottom, end: top})
+
+    function adjustRange(value1: number | null, value2: number | null, bounds: "auto" | [number | null, number | null] | null): [number | null, number | null] {
+      let resolvedValue1 = value1
+      let resolvedValue2 = value2
+
+      if (bounds !== "auto" && (bounds != null)) {
+        const [bound1, bound2] = bounds
+
+        if (typeof value1 === "number" && typeof bound1 === "number") {
```

Comment:
This is technically incorrect, because JS has both `number` and `Number()`. Thus a better way is to use `isNumber()` predicate from `core/util/types`.